### PR TITLE
chore: add integration tests for dedupe.

### LINKED ISSUE #3010: feat: add Integration tests for dedupe
Currently, there are no integration tests to verify if the deduplication feature is working correctly. We need to add tests that confirm the dedupe logic is functioning as intended.

### PR DESCRIPTION
**Pull request by laststylebender14**
**Summary:**  
_Briefly describe the changes made in this PR._

**Issue Reference(s):**  
Fixes #3010

**Build & Testing:**

- [ ] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: tests/core/http.rs [hunk 1]
```diff
@@ -55,6 +55,9 @@ impl Http {
 #[async_trait::async_trait]
 impl HttpIO for Http {
     async fn execute(&self, req: reqwest::Request) -> anyhow::Result<Response<Bytes>> {
+        // TODO: configure this delay only for dedupe requests.
+        let _ = tokio::time::sleep(tokio::time::Duration::from_millis(10)).await;
```
Comment (tusharmath) [inline tests/core/http.rs:59]: Take `latency` from the the specification.

File: tests/execution/test-dedupe.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  name: String
+}
+```
+
+```yml @mock
+- request:
+    method: GET
+    url: http://jsonplaceholder.typicode.com/posts?id=1
+  expectedHits: 1
+  response:
```
Comment (tusharmath) [inline tests/execution/test-dedupe.md:30]: Configure delay here of 10millis like this.
```suggestion
  delay: 10
  response:
```

File: tests/execution/test-dedupe.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      - id: 1
+        userId: 1
+      - id: 2
+        userId: 2
+```
+
+```yml @test
+- method: POST
+  url: http://localhost:8080/graphql
+  concurrent: 10
```
Comment (tusharmath) [inline tests/execution/test-dedupe.md:42]:
```suggestion
  concurrency: 10
```

**Comment by laststylebender14** [comment]
closing in favour of https://github.com/tailcallhq/tailcall/pull/3145

**Pull request closed without merging**